pkg/validate/validate.go

```diff
@@ -279,3 +279,11 @@ func Host(v string) error {
 	}
 	return validateSubdomain(v)
 }
+
+// OnPremClusterName verifies if the cluster name contains a '.' and returns an error if it does.
+func OnPremClusterName(v string) error {
+	if strings.Contains(v, ".") {
+		return errors.New("cluster name must not contain '.'")
```

Comment:
This error should use the kube field package we use for validating install config, e.g.:

```
allErrs = append(allErrs, field.Invalid(field.NewPath("metadata", "name"), c.ObjectMeta.Name, nameErr.Error()))
```

Reply:
You are correct, sir!